### PR TITLE
Add docs page with overview of Gatsby's usage of GraphQL

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

**Comment by m-allanson** [comment]
It's great to see this page getting some ❤️ ! Are you still interested in having a live instance of GraphiQL running in an iframe here? If you are - I'll get around to it at some point.

File: docs/docs/querying-with-graphql.md [hunk 1]
```diff
@@ -1,71 +1,179 @@
...
+    site {
+      siteMetadata {
+        title
+      }
+    }
+  }
+`;
+```
+
+The result of the query is automatically inserted into your React component on the `data` prop. GraphQL and Gatsby lets ask for data and then immediately start using it.
```
Comment (m-allanson) [inline docs/docs/querying-with-graphql.md]: Missing a 'you' here? `GraphQL and Gatsby lets _you_ ask for data`.

**Comment by KyleAMathews** [comment]
Yes! Would love it if you wanted to set that up! Was planning on have ~3 pages for GraphQL. This conceptual overview. Then a reference page (w/ live query examples on gatsbyjs.org). And then finally, a system overview with how Gatsby works overall including the GraphQL/data layer parts of things.

File: docs/docs/querying-with-graphql.md (hunk 1 above)
Comment (KyleAMathews) [reply docs/docs/querying-with-graphql.md]: 👍

**Comment by KyleAMathews** [comment]
Gonna merge this to get it out there and move onto the next PR. Other reviewers, please still review if you'd like and I'll make further PRs to fix things, thanks!

**Pull request merged**